Lib/_pydatetime.py

```diff
@@ -1037,6 +1037,11 @@ def fromtimestamp(cls, t):
         "Construct a date from a POSIX timestamp (like time.time())."
         if t is None:
             raise TypeError("'NoneType' object cannot be interpreted as an integer")
+        if t < 0 and sys.platform.startswith("win"):
```

Comment:
The platform string is always `win32` on Windows:

```suggestion
        if t < 0 and sys.platform == "win32":
```

Reply:
Changed to `os.name == 'nt'` per @ZeroIntensity suggestion.